### DISCUSSION TITLE
Partial parse

### DISCUSSION THREAD

**Discussion by ltenti** [GrapesJS/grapesjs#3711] [General]
Hi everyone,

I'm wondering how to address this kind of scenario:
- I have to allow the users of my application to edit a webpage
- the webpage has some fixed part that the user can't edit
- the whole HTML is produced server-side by a legacy subsystem, but some part inside it has to be freely editable by the user
I realized a custom component that fetches the legacy HTML from the server and "inject" it in the canvas with a model.components(data) statement.
I was thinking about creating a new component type (something like doNotEditMe) with property "editable" set to false and then assigning data-gjs-type="doNotEditMe" to all the elements of the original HTML that has to be not-editable.
Do you think I'm crazy?

**Comment by artf** [comment]
Yeah, unfortunately, if there is no way to recognize the legacy HTML via `isComponent` (example below), then you have to do it in that way.
```js
editor.Components.addType('doNotEditMe', {
  model: {...},
  view: {...},
  isComponent: el => el.classList?.contains('old-el-class'),
})
```

**Comment by dingedi** [comment]
Did you succeed ? Could you give more information because I tried and unfortunately could not define a specifice editable zone in an already existing html code

**Reply by ltenti** [reply]
Hi @dingedi sorry for the late answer!
Yes I succeded following the suggestion from @artf, but, at the end I have to investigate a different approach.

**Reply by carvinlo** [reply]
Hi @ltenti have you tried different approach successfully?
